crates/sui-core/src/consensus_adapter.rs

```diff
@@ -730,9 +745,14 @@ impl ConsensusAdapter {
         let (await_submit, position, positions_moved, preceding_disconnected, amplification_factor) =
             self.await_submit_delay(epoch_store, &transactions[..]);
 
-        // Create the waiter until the transaction is processed by consensus or via checkpoint
-        let processed_via_consensus_or_checkpoint =
-            self.await_consensus_or_checkpoint(transaction_keys.clone(), epoch_store);
+        let processed_via_consensus_or_checkpoint = if must_submit_for_positon {
+            // If we need to get consensus position, don't bypass consensus submission
+            // for tx digest returned from consensus/checkpoint processing
+            future::pending().boxed()
+        } else {
+            self.await_consensus_or_checkpoint(transaction_keys.clone(), epoch_store)
```

Comment:
Out of curiosity, did we observe this being an issue during your tests? For example, skip submitting to consensus because the same tx has been already sequenced via another block position?